Sources/Optimizely+Decide/OptimizelyClient+Decide.swift

```diff
@@ -31,4 +31,144 @@ extension OptimizelyClient {
         return OptimizelyUserContext(optimizely: self, userId: userId, attributes: attributes)
     }
     
+    func decide(user: OptimizelyUserContext,
+                key: String,
+                options: [OptimizelyDecideOption]? = nil) -> OptimizelyDecision {
+        
+        guard let config = self.config else {
+            return OptimizelyDecision.errorDecision(key: key, user: user, error: .sdkNotReady)
+        }
+        
+        guard let feature = config.getFeatureFlag(key: key) else {
+            return OptimizelyDecision.errorDecision(key: key, user: user, error: .featureKeyInvalid(key))
+        }
+        
+        let userId = user.userId
+        let attributes = user.attributes
+        let allOptions = defaultDecideOptions + (options ?? [])
+        let decisionReasons = DecisionReasons()
+        var sentEvent = false
+        var enabled = false
+    
+        let decision = decisionService.getVariationForFeature(config: config,
+                                                              featureFlag: feature,
+                                                              userId: userId,
+                                                              attributes: attributes,
+                                                              options: allOptions)
+        
+        if let featureEnabled = decision?.variation?.featureEnabled {
+            enabled = featureEnabled
+        }
+        
+        var variableMap = [String: Any]()
+        if !allOptions.contains(.excludeVariables) {
+            variableMap = getDecisionVariableMap(feature: feature,
+                                                 variation: decision?.variation,
+                                                 enabled: enabled,
+                                                 reasons: decisionReasons)
+        }
+        
+        var optimizelyJSON: OptimizelyJSON
+        if let opt = OptimizelyJSON(map: variableMap) {
+            optimizelyJSON = opt
+        } else {
+            decisionReasons.addError(OptimizelyError.invalidJSONVariable)
+            optimizelyJSON = OptimizelyJSON.createEmpty()
+        }
+
+        let reasonsToReport = decisionReasons.getReasonsToReport(options: allOptions)
+        
+        if let experimentDecision = decision?.experiment, let variationDecision = decision?.variation {
+            if !allOptions.contains(.disableDecisionEvent) {
+                sendImpressionEvent(experiment: experimentDecision,
+                                    variation: variationDecision,
+                                    userId: userId,
+                                    attributes: attributes)
+                sentEvent = true
```

Comment:
this looks confusing if batchsize is large and we are saying sentEvent is true.

Reply:
sentEvent flag is to mark if an event is supposed to be dispatched (disableDecisionEvent not set, etc). Name changed to "decisionEventDispatched" for clarification.